polling-app-server/src/main/java/com/example/polls/controller/AuthController.java

```diff
@@ -84,18 +102,51 @@
                 signUpRequest.getEmail(), signUpRequest.getPassword());
 
         user.setPassword(passwordEncoder.encode(user.getPassword()));
+        String verificationKey = RandomStringGenerator.getRandom();
+
+        String sha256hex = Hashing.sha256()
+                .hashString(verificationKey, StandardCharsets.UTF_8)
+                .toString();
+
+        user.setVerificationKey(sha256hex);
+
+        String link = "http://localhost:8080/api/auth/verify?verificationKey=" + verificationKey + "&username=" + user.getUsername();
```

Comment:
this link will change at everyone‘s environment，get the baseurl from spring application.properties maybe better

Reply:
u put the username in plain text to verify?

Reply:
If username is exposed, i don't think there is any problem. Usernames are actually visible.

Reply:
> this link will change at everyone‘s environment，get the baseurl from spring application.properties maybe better

Point taken, will update.